sources/packages/backend/apps/api/src/route-controllers/student-scholastic-standings/_tests_/e2e/student-scholastic-standings.institutions.controller.saveScholasticStanding.e2e-spec.ts

```diff
@@ -0,0 +1,321 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeInstitutionLocation,
+  createFakeStudentAppeal,
+  getProviderInstanceForModule,
+  saveFakeApplicationDisbursements,
+} from "@sims/test-utils";
+import {
+  BEARER_AUTH_TYPE,
+  InstitutionTokenTypes,
+  authorizeUserTokenForLocation,
+  createTestingAppModule,
+  getAuthRelatedEntities,
+  getInstitutionToken,
+} from "../../../../testHelpers";
+import * as request from "supertest";
+import {
+  ApplicationStatus,
+  AssessmentTriggerType,
+  InstitutionLocation,
+  StudentAssessmentStatus,
+  StudentScholasticStandingChangeType,
+} from "@sims/sims-db";
+import {
+  APPLICATION_NOT_FOUND,
+  FormNames,
+  FormService,
+  INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+} from "../../../../services";
+import { APPLICATION_CHANGE_NOT_ELIGIBLE } from "../../../../constants";
+import { ScholasticStandingAPIInDTO } from "../../models/student-scholastic-standings.dto";
+import { addDays, getISODateOnlyString } from "@sims/utilities";
+import { AppInstitutionsModule } from "../../../../app.institutions.module";
+import { TestingModule } from "@nestjs/testing";
+
+describe("StudentScholasticStandingsInstitutionsController(e2e)-saveScholasticStanding.", () => {
+  let app: INestApplication;
+  let appModule: TestingModule;
+  let db: E2EDataSources;
+  let collegeFLocation: InstitutionLocation;
+  let institutionUserToken: string;
+  const SCHOLASTIC_STANDING_FORM_NAME =
+    FormNames.ReportScholasticStandingChange;
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appModule = module;
+    db = createE2EDataSources(dataSource);
+    // College F.
+    const { institution: collegeF } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeFLocation = createFakeInstitutionLocation({ institution: collegeF });
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+    // Institution token.
+    institutionUserToken = await getInstitutionToken(
```

Comment:
`getInstitutionToken` is usually retrieved at each test scenario to ensure a valid token.
The method internally has a cache that will ensure the token is not expired. If the token is close to be expired a new one will be retrieved.